src/main/java/com/main/service/impl/PersonServiceImpl.java

```diff
@@ -2,32 +2,50 @@
 
 import java.util.Collections;
 import java.util.List;
+import java.util.stream.Collectors;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
+import com.main.dto.ResponseDto;
 import com.main.entity.Message;
 import com.main.entity.Person;
 import com.main.repository.PersonRepository;
 import com.main.service.PersonService;
 
+@Service
 public class PersonServiceImpl implements PersonService {
 
 	@Autowired
 	private PersonRepository personRepository;
-	
-	
-	 public List<Message> findAllMessagesByPersonId(Long personId) {
-	        // First, find the person by their ID
-	        Person person = personRepository.findById(personId).orElse(null);
-
-	        // If the person is found, retrieve all messages from that person
-	        if (person != null) {
-	            return personRepository.findAllMessagesByPerson(person);
-	        }
-
-	        // Return an empty list if the person is not found
-	        return Collections.emptyList();
-	    }
-	
-	
+
+	public ResponseDto<String> findAllMessagesByPersonId(Long personId, String word) {
+
+		ResponseDto<String> response = new ResponseDto<>();
+
+		// First, find the person by their ID
+		Person person = personRepository.findById(personId).orElse(null);
+
+		// If the person is found, retrieve all messages from that person
+		if (person != null) {
+			List<Message> result = personRepository.findAllMessagesByPersonIdAndSearchWord(personId, word);
+			if (!result.isEmpty()) {
+				response.setStatus(200);
+				response.setMessage("Data retrieved successfully");
+				response.setData(result.stream().map(Message::getContent).collect(Collectors.toList()));
+				return response;
+			} else {
+				response.setStatus(204);
```

Comment:
200  Empty is OK, 204 is used when using POST/PUT/DELETE to indicate the action was performed